lib/system/excpt.nim

```diff
@@ -269,7 +269,7 @@ template addFrameEntry(s: var string, f: StackTraceEntry|PFrame) =
       for i in first..<f.frameMsgLen: add(s, frameMsgBuf[i])
   add(s, "\n")
 
-proc `$`(stackTraceEntries: seq[StackTraceEntry]): string =
+proc `$`(stackTraceEntries: seq[StackTraceEntry]): string {.sideEffect.} =
```

Comment:
why should this be considered have a side effect? isn't it a case where the things getting called need to be changed to noSideEffect?

Reply:
See the discussion in https://github.com/nim-lang/Nim/pull/19620#issuecomment-1074076148
iirc, we either lie about side effects when `-d:nimStackTraceOverride` is on (#19620), or assume that there can always be side effects (this PR)

Or find a better system than `nimStackTraceOverride` altogether

Reply:
seems to me the analysis in #19620 is wrong - accessing _constant_ global data is fine, even under no-side-effect - the mapping from address to line is a good example for this - the important bit behind "no side effect" is that a repeated call with the same input will always yield the same output which is true in this case